test/src/loaders/workspace_uploader/workspace_uploader_test.py

```diff
@@ -223,78 +237,220 @@ def test_post_process(setup_and_teardown):
+    asu.save_assemblies_from_fastas.assert_called_once_with(
         {
-            "file": {"path": "/path/to/file/in/AssembilyUtil"},
             "workspace_id": 12345,
-            "assembly_name": assembly_name,
+            "inputs": [
+                {
+                    "file": assembly_tuple.container_internal_assembly_path,
+                    "assembly_name": assembly_tuple.assembly_name,
+                    "object_metadata": {"load_id": "214"},
+                }
+            ]
         }
     )
 
 
-def test_assembly_files_in_parallel(setup_and_teardown):
+def test_generator(setup_and_teardown):
+    params = setup_and_teardown
+    assembly_dirs = params.assembly_dirs
+    wait_to_upload_assemblies = {
+        assembly_name: assembly_dir
+        for assembly_name, assembly_dir in zip(ASSEMBLY_NAMES, assembly_dirs)
+    }
+    assemblyTuple_list = list(workspace_uploader._gen(wait_to_upload_assemblies, 1))
+    expected_assemblyTuple_list = [
+        [
+            workspace_uploader._AssemblyTuple(
+                "GCF_000979855.1_gtlEnvA5udCFS_genomic.fna.gz",
+                assembly_dirs[0],
+                "/kb/module/work/tmp/GCF_000979855.1_gtlEnvA5udCFS_genomic.fna.gz",
+            )
+        ],
+        [
+            workspace_uploader._AssemblyTuple(
+                "GCF_000979175.1_gtlEnvA5udCFS_genomic.fna.gz",
+                assembly_dirs[1],
+                "/kb/module/work/tmp/GCF_000979175.1_gtlEnvA5udCFS_genomic.fna.gz",
+            )
+        ],
+    ]
+    assert assemblyTuple_list == expected_assemblyTuple_list
+
+
+def test_upload_assembly_files_in_parallel(setup_and_teardown):
     params = setup_and_teardown
+    src_files = params.target_files
+    assembly_dirs = params.assembly_dirs
     upload_dir = Path(params.tmp_dir) / "upload_dir"
     upload_dir.mkdir()
-    assembly_dirs = params.assembly_dirs
+    output_dir = Path(params.tmp_dir) / "output_dir"
+    output_dir.mkdir()
 
     wait_to_upload_assemblies = {
         assembly_name: assembly_dir
         for assembly_name, assembly_dir in zip(ASSEMBLY_NAMES, assembly_dirs)
     }
 
-    conf = Mock()
-    conf.workers = 5
-    conf.input_queue = Queue()
-    conf.output_queue = Queue()
-
-    # an uploaded successful
-    conf.output_queue.put((ASSEMBLY_NAMES[0], "12345_58_1"))
-    # an upload failed
-    conf.output_queue.put((ASSEMBLY_NAMES[1], None))
-
-    failed_names = workspace_uploader._upload_assembly_files_in_parallel(
-        conf, "CI", 12345, upload_dir, wait_to_upload_assemblies
-    )
+    ws = create_autospec(Workspace, spec_set=True, instance=True)
+    asu = create_autospec(AssemblyUtil, spec_set=True, instance=True)
+    asu.save_assemblies_from_fastas.return_value = {
+        "results":[
+            {"upa": "12345/58/1"},
+            {"upa": "12345/60/1"}
+        ]
+    }
 
-    expected_tuple1 = (
+    uploaded_count = workspace_uploader._upload_assembly_files_in_parallel(
+        asu,
```

Comment:
Seems like there should be one or more asserts that asu was called correctly

Reply:
👍

Reply:
:+1: